esp-hal/src/uart.rs

```diff
@@ -949,14 +981,19 @@ where
     }
 
     #[cfg(any(esp32, esp32s2))]
-    fn change_baud_internal(&self, baudrate: u32, clocks: &Clocks) {
-        // we force the clock source to be APB and don't use the decimal part of the
-        // divider
-        let clk = clocks.apb_clock.to_Hz();
+    fn change_baud_internal(&self, baudrate: u32, clock_source: ClockSource, clocks: &Clocks) {
+        let clk = match clock_source {
+            ClockSource::Apb => clocks.apb_clock.to_Hz(),
+            ClockSource::RefTick => 1, // ESP32 TRM, section 3.2.4.2 (RefTick)
```

Comment:
Is it the same for S2 too?

Reply:
Yep, but will update the comment with S2 section
> Normally, one REF_TICK cycle lasts for 1 μs (1MHz) - esp32s2 trm